frontend/packages/console-plugin-sdk/src/typings/dashboards.ts

```diff
@@ -373,9 +374,10 @@ export type PrometheusHealthHandler = (
   additionalResource?: FirehoseResult<K8sResourceKind | K8sResourceKind[]>,
 ) => SubsystemHealth;
 
-export type CRDHealthHandler = (
-  response: FirehoseResult<K8sResourceKind | K8sResourceKind[]>,
-  additionalResource?: FirehoseResult<K8sResourceKind | K8sResourceKind[]>,
+export type ResourceHealthHandler = (
```

Comment:
we could just pass result of fetching like 

```suggestion
export type ResourceHealthHandler<R extends ResourcesObject> = (results: WatchK8sResults<R>) => SubsystemHealth;
```

`WatchK8sResults` will need to be exported from `k8s-watch-hook`